modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveByPatternProcessor.java

```diff
@@ -0,0 +1,186 @@
+                if (!metadataFields.contains(field)) {
+                    final boolean matched = excludeFieldPatterns.stream().anyMatch(pattern -> Regex.simpleMatch(pattern, field));
+                    if (!matched) {
+                        document.removeField(field);
+                    }
+                }
+            });
+        }
+
+        return document;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        public Factory() {}
+
+        @Override
+        public RemoveByPatternProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+        ) throws Exception {
+            final List<String> fieldPatterns = new ArrayList<>();
+            final List<String> excludeFieldPatterns = new ArrayList<>();
+            final Object fieldPattern = ConfigurationUtils.readOptionalObject(config, "field_pattern");
+            final Object excludeFieldPattern = ConfigurationUtils.readOptionalObject(config, "exclude_field_pattern");
+
+            if (fieldPattern == null && excludeFieldPattern == null || fieldPattern != null && excludeFieldPattern != null) {
+                throw newConfigurationException(
+                    TYPE,
+                    processorTag,
+                    "field_pattern",
+                    "ether field_pattern or exclude_field_pattern must be set"
+                );
+            }
+
+            if (fieldPattern != null) {
+                if (fieldPattern instanceof List) {
+                    @SuppressWarnings("unchecked")
+                    List<String> fieldPatternList = (List<String>) fieldPattern;
+                    fieldPatterns.addAll(fieldPatternList);
+                } else {
+                    fieldPatterns.add((String) fieldPattern);
+                }
+                validateFieldPatterns(processorTag, fieldPatterns, "field_pattern");
+                return new RemoveByPatternProcessor(processorTag, description, fieldPatterns, null);
+            } else {
+                if (excludeFieldPattern instanceof List) {
+                    @SuppressWarnings("unchecked")
+                    List<String> excludeFieldPatternList = (List<String>) excludeFieldPattern;
+                    excludeFieldPatterns.addAll(excludeFieldPatternList);
+                } else {
+                    excludeFieldPatterns.add((String) excludeFieldPattern);
+                }
+                validateFieldPatterns(processorTag, excludeFieldPatterns, "exclude_field_pattern");
+                return new RemoveByPatternProcessor(processorTag, description, null, excludeFieldPatterns);
+            }
+        }
+
+        private void validateFieldPatterns(String processorTag, List<String> patterns, String patternKey) {
```

Comment:
What if the field pattern contains 2 asterix : Eg: `*ab*b` ?

Reply:
It's valid, any filed name match that pattern will be removed or kept.

Reply:
I've added this case to the yml test file.